extensions/redis-client/runtime/src/main/java/io/quarkus/redis/client/runtime/RedisAPIProducer.java

```diff
@@ -1,137 +1,55 @@
 package io.quarkus.redis.client.runtime;
 
-import java.net.InetSocketAddress;
-import java.util.Set;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
 
 import javax.annotation.PreDestroy;
-import javax.enterprise.context.ApplicationScoped;
-import javax.enterprise.inject.Produces;
 
 import io.quarkus.redis.client.RedisClient;
 import io.quarkus.redis.client.reactive.ReactiveRedisClient;
-import io.quarkus.runtime.configuration.ConfigurationException;
 import io.vertx.core.Vertx;
 import io.vertx.redis.client.Redis;
 import io.vertx.redis.client.RedisAPI;
-import io.vertx.redis.client.RedisClientType;
 import io.vertx.redis.client.RedisOptions;
 
 class RedisAPIProducer {
```

Comment:
Is there a reason why we are not being consistent and use the synthetic bean pattern for this too?

Reply:
You mean drop this build item  https://github.com/quarkusio/quarkus/blob/6a7ba55365abf623a196fe106800c490c88fe6ac/extensions/redis-client/deployment/src/main/java/io/quarkus/redis/client/deployment/RedisClientProcessor.java#L59-L64  and use a synthetic bean instead? 

If so, I saw no need of the synthetic bean for this one.

Reply:
OK, we can live with it I suppose.